src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -4,34 +4,81 @@
 #include "software/ai/hl/stp/tactic/dribble/dribble_fsm.h"
 #include "software/geom/algorithms/contains.h"
 #include "software/geom/algorithms/distance.h"
+#include "software/geom/algorithms/step_along_perimeter.h"
 #include "software/geom/stadium.h"
 
 std::optional<Point> CreaseDefenderFSM::findBlockThreatPoint(
     const Field& field, const Point& enemy_threat_origin,
     const TbotsProto::CreaseDefenderAlignment& crease_defender_alignment,
-    double robot_obstacle_inflation_factor)
+    double robot_obstacle_inflation_factor, double x_shift_meters)
 {
+    // Look directly at the point
+    // Return the segments that form the path around the crease that the
+    // defenders must follow. It's basically the crease inflated by one robot radius
+    // multiplied by a factor
+    double robot_radius_expansion_amount =
+        ROBOT_MAX_RADIUS_METERS * robot_obstacle_inflation_factor;
+    Rectangle inflated_defense_area =
+        field.friendlyDefenseArea().expand(robot_radius_expansion_amount);
```

Comment:
This code is duplicated elsewhere, please make a note/minor issue that for post RoboCup (if you would like to do it later) it should be extracted into a function which returns the rectangle. Ideally, this shape should be a constant, but since the field size is technically dynamic a function is required instead.

Reply:
fixed